src/checkqueue.h

```diff
@@ -5,179 +5,437 @@
 #ifndef BITCOIN_CHECKQUEUE_H
 #define BITCOIN_CHECKQUEUE_H
 
-#include <algorithm>
 #include <vector>
+#include "util.h"
+#include "utiltime.h"
+#include <atomic>
+#include <thread>
+#include <chrono>
+#include <sstream>
+#include <iostream>
+// This should be ignored eventually, but needs testing to ensure this works on more platforms
+static_assert(ATOMIC_BOOL_LOCK_FREE, "shared_status not lock free");
```

Comment:
I added a commit which changes them from static_asserts to warnings. I'm unsure if the warning is properly x-platform; seems to not be a consistent thing. In any case, not being lockfree only effects performance, not correctness.